app/celery/tasks.py

```diff
@@ -171,60 +169,14 @@ def choose_database_queue(template: Any, service: Service):
         return QueueNames.NORMAL_DATABASE
 
 
-def process_row(row: Row, template: Template, job: Job, service: Service):
-    template_type = template.template_type
-    client_reference = row.get("reference")
-    signed = signer.sign(
-        {
-            "api_key": job.api_key_id and str(job.api_key_id),
-            "template": str(template.id),
-            "template_version": job.template_version,
-            "job": str(job.id),
-            "to": row.recipient,
-            "row_number": row.index,
-            "personalisation": dict(row.personalisation),
-            "queue": queue_to_use(job.notification_count),
-            "client_reference": client_reference.data if client_reference else None,
-        }
-    )
-
-    notification_id = create_uuid()
-
-    sender_id = str(job.sender_id) if job.sender_id else None
-
-    send_fns = {SMS_TYPE: save_sms, EMAIL_TYPE: save_email, LETTER_TYPE: save_letter}
-
-    send_fn = send_fns[template_type]
-
-    task_kwargs = {}
-    if sender_id:
-        task_kwargs["sender_id"] = sender_id
-
-    # the same_sms and save_email task are going to be using template and service objects from cache
-    # these objects are transient and will not have relationships loaded
-    if service_allowed_to_send_to(row.recipient, service, KEY_TYPE_NORMAL):
-        send_fn.apply_async(
-            (
-                str(service.id),
-                notification_id,
-                signed,
-            ),
-            task_kwargs,
-            queue=choose_database_queue(template, service),
-        )
-    else:
-        current_app.logger.debug("SMS {} failed as restricted service".format(notification_id))
-
-
 def process_rows(rows: List, template: Template, job: Job, service: Service):
```

Comment:
could use `List[Row]` to be more specific.

Reply:
ah but in the case of the test its notification. going to leave this as is

Reply:
😬